### PR TITLE
[DO NOT MERGE] fix(core): Force-wake the waker if an eager poll is Pending

### PR DESCRIPTION
**Pull request by mmastrac**

### CODE REVIEW

**Comment by aapoalas** [comment]
Am I correct in presuming that this makes the event loop light up all happy-like during a long-polling async op?

**Comment by bartlomieju** [comment]
> Am I correct in presuming that this makes the event loop light up all happy-like during a long-polling async op?

No, it's only woken up once per op if the op in not immediately ready.

**Comment by aapoalas** [comment]
> > Am I correct in presuming that this makes the event loop light up all happy-like during a long-polling async op?
> 
> No, it's only woken up once per op if the op in not immediately ready.

That's good then!

**Review by aapoalas — APPROVED** [review]

**Review by bartlomieju — CHANGES_REQUESTED** [review]
Not ready, needs a reproducible test before we land it.

File: core/runtime/ops.rs [hunk 1]
```diff
@@ -120,7 +120,13 @@ pub fn queue_async_op<'s>(
   let mut pinned = op.map(move |res| (promise_id, id, res)).boxed_local();
 
   match pinned.poll_unpin(&mut Context::from_waker(noop_waker_ref())) {
-    Poll::Pending => {}
+    Poll::Pending => {
+      // TODO(mmastrac): We are polling a future directly, and then shoving it into a task where it gets polled
+      // in a completely different way. While this shouldn't necessarily be an issue, this appears to cause a loss
+      // of wakeup events in certain situations that are not entirely clear. For now we're going to force the event
+      // loop to wake up which will then poll the JoinSet, correct the wakers, and ensures that we don't lose events.
+      ctx.state.borrow_mut().waker.wake();
```
Comment (piscisaureus) [inline core/runtime/ops.rs:128]: Would it not make more sense to pass `ctx.state.waker` to the `poll_unpin()` call above?

Comment (bartlomieju) [reply core/runtime/ops.rs:128]: AFAIK there's no way to get `&Waker` from `AtomicWaker`

**Comment by mmastrac** [comment]
Closing this for now

**Pull request closed without merging**